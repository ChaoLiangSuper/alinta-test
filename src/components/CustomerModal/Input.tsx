import React from 'react';
import styled from 'styled-components';

interface StyledInputProps {
  invalid?: boolean;
}

interface InputProps extends StyledInputProps {
  name: string;
  label: string;
  date?: boolean;
  value: string;
  onChange: (value: string) => void;
}

const StyledLabel = styled.label`
  display: flex;
  width: 100%;
`;

const StyledText = styled.span`
  flex-grow: 2;
  flex-basis: 0;
`;

const StyledInput = styled.input<StyledInputProps>`
  flex-grow: 3;
  flex-basis: 0;
  width: 100%;
  border: ${({ invalid }) => (invalid ? '1px solid red' : '1px solid grey')};
  border-radius: ${({ theme }) => theme.borderRadius()};
`;

const Input: React.FC<InputProps> = ({
  name,
  label,
  date,
  value,
  onChange,
  invalid,
}) => {
  return (
    <StyledLabel>
      <StyledText>{label}</StyledText>
      <StyledInput
        autoComplete='off'
        name={name}
        type={date ? 'date' : 'text'}
        value={value}
        invalid={invalid}
        onChange={({ target }) => onChange(target.value)}
      />
    </StyledLabel>
  );
};

export default Input;
