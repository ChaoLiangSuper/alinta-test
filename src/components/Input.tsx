import React from 'react';
import styled from 'styled-components';

interface StyledInputProp {
  invalid?: boolean;
}

interface InputProp extends StyledInputProp {
  name: string;
  label: string;
  date?: boolean;
  value: string;
  onChange: (value: string) => void;
}

const StyledLabel = styled.label`
  display: flex;
  width: 100%;
`;

const StyledText = styled.span`
  flex-grow: 2;
  flex-basis: 0;
`;

const StyledInput = styled.input<StyledInputProp>`
  flex-grow: 3;
  flex-basis: 0;
  width: 100%;
  border: ${({ invalid }) => (invalid ? '1px solid red' : '1px solid grey')};
  border-radius: ${({ theme }) => theme.borderRadius()};
`;

const Input: React.FC<InputProp> = ({
  name,
  label,
  date,
  value,
  onChange,
  invalid,
}) => {
  return (
    <StyledLabel>
      <StyledText>{label}</StyledText>
      <StyledInput
        name={name}
        type={date ? 'date' : 'text'}
        value={value}
        invalid={invalid}
        onChange={({ target }) => onChange(target.value)}
      />
    </StyledLabel>
  );
};

export default Input;
