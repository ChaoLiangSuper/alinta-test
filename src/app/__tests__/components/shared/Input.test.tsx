import React from 'react';
import Input, { StyledInput } from '../../../components/shared/Input';
import { shallow, ShallowWrapper } from 'enzyme';

describe('<Input />', () => {
  let wrapper: ShallowWrapper;
  const mockFunc = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Input onChange={mockFunc} />);
  });

  it('should render without error', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle change event', () => {
    const textInput = wrapper.find(StyledInput);
    textInput.simulate('change', { target: { value: 'text' } });
    expect(mockFunc).toHaveBeenCalledWith('text');
  });
});
