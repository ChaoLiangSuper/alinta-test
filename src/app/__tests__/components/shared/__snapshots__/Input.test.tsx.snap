// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render without error 1`] = `
<styled.label>
  <styled.input
    autoComplete="off"
    onChange={[Function]}
    type="text"
  />
</styled.label>
`;
